### PR TITLE
Issue 18563 create resource for bundle operations

### PR DESCRIPTION
**Pull request by dsilvam**
This PR includes the following:

* New Resource for uploading Bundles (POST multipart api/bundle) with options for sync/async upload
* Postman tests for resource above
* Included the rest of properties for the GraphQL Page API, for the `page` properties
* Postman tests for the above 
* Make all the Postman tests for PageResource to pass by: 
	- Importing a bundle with the context (pages to test)
	- Fixing all the urls 
	- Fixing all the asserts 
	- Including a bunch of new asserts
	- Renaming and reorganizing tests 
* Validation to avoid NPE when `contentletsIds` was not sent to REST Page Layout API
* Better exception message for non existing pages on Page Resource REST

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java [hunk 1]
```diff
@@ -588,4 +620,71 @@ public Response deleteAllSuccess(@Context final HttpServletRequest request,
                 "Removing bundles in a separated process, the result of the operation will be notified")).build();
     } // deleteAllSuccess.
 
+    @POST
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML, MediaType.TEXT_PLAIN})
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public final Response postMultipart(@Context final HttpServletRequest request, @Context final HttpServletResponse response,
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:628]: postman test?

**Review by freddyucv — CHANGES_REQUESTED** [review]

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java [hunk 2]
```diff
@@ -588,4 +620,71 @@ public Response deleteAllSuccess(@Context final HttpServletRequest request,
...
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requiredBackendUser(true)
+                .requiredFrontendUser(false)
+                .requestAndResponse(request, response)
+                .rejectWhenNoUser(true)
+                .init();
+
+        for (final BodyPart part : multipart.getBodyParts()) {
+
+            InputStream inputStream = (part.getEntity() instanceof InputStream) ? InputStream.class
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:641]: Codacy found an issue: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java [hunk 3]
```diff
@@ -588,4 +620,71 @@ public Response deleteAllSuccess(@Context final HttpServletRequest request,
...
+
+            if (inputStream == null) {
+                continue;
+            }
+            final ContentDisposition meta = part.getContentDisposition();
+            if (meta == null) {
+                continue;
+            }
+            final String fileName = meta.getFileName();
+            if (fileName == null || fileName.startsWith(".") || fileName.contains("/.")) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:653]: Codacy found an issue: [This call to String.startsWith can be rewritten using String.charAt(0)](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:641]: Codacy found an issue: [Ensure that resources like this boolean object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:641]: Codacy found an issue: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:653]: Codacy found an issue: [This call to String.startsWith can be rewritten using String.charAt(0)](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:641]: Codacy found an issue: [Ensure that resources like this boolean object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:641]: Codacy found an issue: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:653]: Codacy found an issue: [This call to String.startsWith can be rewritten using String.charAt(0)](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:641]: Codacy found an issue: [Ensure that resources like this boolean object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:641]: Codacy found an issue: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:653]: Codacy found an issue: [This call to String.startsWith can be rewritten using String.charAt(0)](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:641]: Codacy found an issue: [Ensure that resources like this boolean object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java [hunk 4]
```diff
@@ -406,12 +406,12 @@ public final Response addContent(@Context final HttpServletRequest request,
         Logger.debug(this, ()->String.format("Saving page's content: %s",
                 pageContainerForm != null ? pageContainerForm.getRequestJson() : null));
 
+        final InitDataObject initData = webResource.init(request, response,true);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java:409]: Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java (hunk 1 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/rest/BundleResource.java]: Done

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java [hunk 5]
```diff
@@ -588,4 +620,71 @@ public Response deleteAllSuccess(@Context final HttpServletRequest request,
...
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requiredBackendUser(true)
+                .requiredFrontendUser(false)
+                .requestAndResponse(request, response)
+                .rejectWhenNoUser(true)
+                .init();
+
+        for (final BodyPart part : multipart.getBodyParts()) {
+
+            InputStream inputStream = (part.getEntity() instanceof InputStream) ? (InputStream) part
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:641]: Codacy found an issue: [Ensure that resources like this boolean object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java [hunk 6]
```diff
@@ -588,4 +620,71 @@ public Response deleteAllSuccess(@Context final HttpServletRequest request,
...
+
+            if (inputStream == null) {
+                continue;
+            }
+            final ContentDisposition meta = part.getContentDisposition();
+            if (meta == null) {
+                continue;
+            }
+            final String fileName = meta.getFileName();
+            if (UtilMethods.isNotSet(fileName) || fileName.startsWith(".") || fileName.contains("/.")) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:653]: Codacy found an issue: [This call to String.startsWith can be rewritten using String.charAt(0)](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java [hunk 7]
```diff
@@ -27,13 +27,15 @@
...
 import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.exception.DoesNotExistException;
 import com.dotmarketing.exception.DotDataException;
 import com.dotmarketing.exception.DotSecurityException;
 import com.dotmarketing.filters.CMSUrlUtil;
 import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
 import com.dotmarketing.portlets.contentlet.model.Contentlet;
 import com.dotmarketing.portlets.contentlet.model.ContentletVersionInfo;
 import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.portlets.htmlpageasset.business.render.HTMLPageAssetNotFoundException;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java:38]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.htmlpageasset.business.render.HTMLPageAssetNotFoundException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java [hunk 8]
```diff
@@ -406,12 +407,12 @@ public final Response addContent(@Context final HttpServletRequest request,
...
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java:410]: Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:641]: Codacy found an issue: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:641]: Codacy found an issue: [Ensure that resources like this boolean object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java:410]: Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java:38]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.htmlpageasset.business.render.HTMLPageAssetNotFoundException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:641]: Codacy found an issue: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:653]: Codacy found an issue: [This call to String.startsWith can be rewritten using String.charAt(0)](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java:410]: Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java:38]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.htmlpageasset.business.render.HTMLPageAssetNotFoundException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:641]: Codacy found an issue: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:653]: Codacy found an issue: [This call to String.startsWith can be rewritten using String.charAt(0)](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:641]: Codacy found an issue: [Ensure that resources like this boolean object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java:410]: Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java [hunk 9]
```diff
@@ -588,4 +621,78 @@ public Response deleteAllSuccess(@Context final HttpServletRequest request,
...
+
+                if (inputStream == null) {
+                    continue;
+                }
+                final ContentDisposition meta = part.getContentDisposition();
+                if (meta == null) {
+                    continue;
+                }
+                final String fileName = meta.getFileName();
+                if (UtilMethods.isNotSet(fileName) || fileName.startsWith(".") || fileName
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:654]: Codacy found an issue: [This call to String.startsWith can be rewritten using String.charAt(0)](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java:410]: Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java [hunk 10]
```diff
@@ -588,4 +621,77 @@ public Response deleteAllSuccess(@Context final HttpServletRequest request,
...
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:653]: Codacy found an issue: [This call to String.startsWith can be rewritten using String.charAt(0)](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java:410]: Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java (hunk 10 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:653]: Codacy found an issue: [This call to String.startsWith can be rewritten using String.charAt(0)](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:653]: Codacy found an issue: [This call to String.startsWith can be rewritten using String.charAt(0)](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java:410]: Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java:410]: Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java (hunk 10 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:653]: Codacy found an issue: [This call to String.startsWith can be rewritten using String.charAt(0)](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/BundleResource.java (hunk 10 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/BundleResource.java:653]: Codacy found an issue: [This call to String.startsWith can be rewritten using String.charAt(0)](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java:410]: Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)

**Pull request merged**